CourseSketchProjects/IdentityServer/src/main/java/coursesketch/database/identity/IdentityManager.java

```diff
@@ -0,0 +1,549 @@
+import com.mongodb.DBObject;
+import com.mongodb.MongoClient;
+import coursesketch.database.auth.AuthenticationException;
+import coursesketch.database.auth.AuthenticationResponder;
+import coursesketch.database.auth.Authenticator;
+import coursesketch.database.interfaces.AbstractCourseSketchDatabaseReader;
+import coursesketch.server.authentication.HashManager;
+import coursesketch.server.interfaces.AbstractServerWebSocketHandler;
+import coursesketch.server.interfaces.ServerInfo;
+import database.DatabaseAccessException;
+import database.DatabaseStringConstants;
+import org.bson.types.ObjectId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import protobuf.srl.school.School;
+import protobuf.srl.services.authentication.Authentication;
+
+import java.nio.charset.StandardCharsets;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static database.DbSchoolUtility.getCollectionFromType;
+import static database.DbSchoolUtility.getParentItemType;
+
+/**
+ * Created by dtracers on 10/7/2015.
+ */
+public final class IdentityManager extends AbstractCourseSketchDatabaseReader implements IdentityManagerInterface {
+
+    /**
+     * Declaration and Definition of Logger.
+     */
+    private static final Logger LOG = LoggerFactory.getLogger(IdentityManager.class);
+
+    /**
+     * The database that the auth checker grabs data from.
+     */
+    private DB database;
+
+    /**
+     * Creates An IdentityManager with a database.
+     *
+     * @param database
+     *         The database where all of the data is stored.
+     */
+    public IdentityManager(final DB database) {
+        super(null);
+        this.database = database;
+    }
+
+    /**
+     * Creates An Identity Manager with a server information.
+     *
+     * @param serverInfo
+     *         The information about the database location.
+     */
+    public IdentityManager(final ServerInfo serverInfo) {
+        super(serverInfo);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @throws DatabaseAccessException
+     *         thrown if the database already exist when the database is created.
+     */
+    @Override protected void onStartDatabase() throws DatabaseAccessException {
+        if (this.database != null) {
+            throw new DatabaseAccessException("Mongo instance already exists!");
+        }
+        final MongoClient mongoClient = new MongoClient(super.getServerInfo().getDatabaseUrl());
+        this.database = mongoClient.getDB(super.getServerInfo().getDatabaseName());
+        final DBCollection collection = this.database.getCollection(DatabaseStringConstants.USER_COLLECTION);
+        collection.createIndex(new BasicDBObject(DatabaseStringConstants.USER_NAME, 1));
+        super.setDatabaseStarted();
+    }
+
+    /**
+     * {@inheritDoc}
```

Comment:
the inheritDoc means that this data is documented in its parent class and does not need to be documented here